ethereum/eip712/signed_data.go

```diff
@@ -0,0 +1,1053 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package eip712
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/consensus/clique"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type SigFormat struct {
+	Mime        string
+	ByteVersion byte
+}
+
+var (
+	IntendedValidator = SigFormat{
+		accounts.MimetypeDataWithValidator,
+		0x00,
+	}
+	DataTyped = SigFormat{
+		accounts.MimetypeTypedData,
+		0x01,
+	}
+	ApplicationClique = SigFormat{
+		accounts.MimetypeClique,
+		0x02,
+	}
+	TextPlain = SigFormat{
+		accounts.MimetypeTextPlain,
+		0x45,
+	}
+)
+
+type ValidatorData struct {
+	Address common.Address
+	Message hexutil.Bytes
+}
+
+type TypedData struct {
```

Comment:
I think can import this types from geth https://github.com/ethereum/go-ethereum/blob/5a0d487c3bd8f3a05a41ffde1b20b17567abd029/signer/core/apitypes/types.go#L201 to reduce mantainability

Reply:
If you can make it work, it would be great. I know that when I started importing some of the code, I had some problems with 2 dependencies that were conflicting with each other.

Reply:
Just pushed the fix for that! You need to merge the latest from main since we updated geth. They made the types more accesible